packages/test-utils/src/cli.ts

```diff
@@ -0,0 +1,101 @@
+import childProcess from "node:child_process";
+import type {Argv} from "yargs";
+import {wrapTimeout} from "./timeout.js";
+import {nodeJsBinaryPath, repoRootPath, tsNodeBinaryPath} from "./path.js";
+import {
+  ExecChildProcessOptions,
+  SpawnChildProcessOptions,
+  execChildProcess,
+  spawnChildProcess,
+} from "./childProcess.js";
+
+// We need to make it easy for the user to pass the args for the CLI
+// yargs treat `["--preset minimal"] as a single arg, so we need to split it ["--preset", "minimal"]
+function parseArgs(args: string[]): string[] {
+  return args.map((a) => a.split(" ")).flat();
+}
+
+type CommandRunOptions = {
+  timeoutMs: number;
+};
+
+/**
+ * Run the cli command inside the main process from the Yargs object
+ */
+export async function runCliCommand<T>(
+  cli: Argv<T>,
+  args: string[],
+  opts: CommandRunOptions = {timeoutMs: 1000}
+): Promise<string> {
+  return wrapTimeout(
+    // eslint-disable-next-line no-async-promise-executor
+    new Promise(async (resolve, reject) => {
+      await cli.parseAsync(parseArgs(args), {}, (err, _argv, output) => {
+        if (err) return reject(err);
+
+        resolve(output);
+      });
+    }),
+    opts.timeoutMs
+  );
+}
+
+/**
+ * Exec a command in bash script mode. Useful for short-running commands
+ *
+ * @param command - The command should be relative to mono-repo root
+ * @param args
+ * @param opts
+ * @returns
+ */
+export function execCliCommand(
+  command: string,
+  args: string[],
+  opts?: ExecChildProcessOptions & {runWith?: "node" | "ts-node"}
+): Promise<string> {
+  const commandPrefixed =
+    opts?.runWith === "ts-node"
+      ? // ts-node --esm cli.ts
+        tsNodeBinaryPath
+      : // node cli.js
+        nodeJsBinaryPath;
+
+  const argsPrefixed =
+    opts?.runWith === "ts-node"
+      ? // ts-node --esm cli.ts
+        ["--esm", repoRootPath(command), ...args]
+      : // node cli.js
+        [repoRootPath(command), ...args];
+
+  return execChildProcess([commandPrefixed, ...parseArgs(argsPrefixed)], opts);
+}
+
+/**
+ * Spawn a process and keep it running
+ *
+ * @param command - The command should be relative to mono-repo root
+ * @param args
+ * @param opts
+ * @returns
+ */
+export async function spawnCliCommand(
+  command: string,
+  args: string[],
+  opts?: SpawnChildProcessOptions & {runWith?: "node" | "ts-node"}
```

Comment:
we can't run with ts-node anymore, we have to do `node --loader ts-node/esm`

Reply:
It was running with `ts-node`, though now I have changed the implementation to use `node --loader ts-node/esm`.